client/src/components/TabPane/Cards/MessagesCard.js

```diff
@@ -1,7 +1,462 @@
+          resetEditing();
+        }}
+        onOk={saveEdited}
+      >
+        <Select
+          style={{
+            width: '100%',
+          }}
+          value={editingMessage?.language}
+          placeholder="Please add language"
+          onChange={(e) => {
+            setEditingMessage((pre) => {
+              return { ...pre, language: e };
+            });
+          }}
+          options={langData}
+        />
+        <br/>
+        <Select
+          style={{
+            width: '100%',
+          }}
+          value={editingMessage?.original_message}
+          placeholder="Please add original message"
+          onChange={(e) => {
+            setEditingMessage((pre) => {
+              return { ...pre, original_message: e };
+            });
+          }}
+          options={originalMessData}
+        />
+        <br/>
+        <Select
+          mode="multiple"
+          allowClear
+          style={{
+            width: '100%',
+          }}
+          value={editingMessage?.tags}
+          placeholder="Please select tags"
+          onChange={(e) => {
+            setEditingMessage((pre) => {
+              console.log(e);
+              return { ...pre, tags: e };
+            });
+          }}
+          options={tagData}
+        />
+        <Input
+          value={editingMessage?.content}
+          placeholder="Message content"
+          onChange={(e) => {
+            setEditingMessage((pre) => {
+              return { ...pre, content: e.target.value };
+            });
+          }}
+        ></Input>
+      </Modal>
+
+      <Modal
+        title="Add new message"
+        open={isAdding}
+        okText="Save"
+        onCancel={() => {
+          resetAdding();
+        }}
+        onOk={saveNewMessage}
+      >
+        <Select
+          style={{
+            width: '100%',
+          }}
+          value={addingMessage?.language}
+          placeholder="Please add language"
+          onChange={(e) => {
+            setAddingMessage((pre) => {
+              return { ...pre, language: e };
+            });
+          }}
+          options={langData}
+        />
+        <br/>
+        <Select
+          style={{
+            width: '100%',
+          }}
+          value={addingMessage?.original_message}
+          placeholder="Please add original message"
+          onChange={(e) => {
+            setAddingMessage((pre) => {
+              return { ...pre, original_message: e };
+            });
+          }}
+          options={originalMessData}
+        />
+        <br/>
+        <Select
+          mode="multiple"
+          allowClear
+          style={{
+            width: '100%',
+          }}
+          value={addingMessage?.tags}
+          placeholder="Please select tags"
+          onChange={(e) => {
+            setAddingMessage((pre) => {
+              return { ...pre, tags: e };
+            });
+          }}
+          options={tagData}
+        />
+        <Input
```

Comment:
Please change this component to _Textarea_, maybe even _Textarea with character counting_

Reply:
Good idea, done.